bazel/foreign_cc/vpp_vcl.patch

```diff
@@ -49,3 +53,145 @@
    target_sources(${lib} PRIVATE $<TARGET_OBJECTS:${lo}>)
  
    if(VPP_LIB_VERSION)
+diff --git src/tools/vppapigen/vppapigen.py src/tools/vppapigen/vppapigen.py
+index 8415c28fb..a017d9fc6 100755
+--- src/tools/vppapigen/vppapigen.py
++++ src/tools/vppapigen/vppapigen.py
+@@ -7,6 +7,13 @@ import logging
+ import binascii
+ import os
+ from subprocess import Popen, PIPE
++
++# Put ply on the path ...
++plypath = os.path.join(
++    os.environ["EXT_BUILD_ROOT"],
++    os.path.dirname(os.environ["PLYPATHS"].split()[0]))
++sys.path += [plypath]
++
+ import ply.lex as lex
+ import ply.yacc as yacc
+ 
+diff --git src/vat/CMakeLists.txt src/vat/CMakeLists.txt
+index e5945b20d..3d76f3d88 100644
+--- src/vat/CMakeLists.txt
++++ src/vat/CMakeLists.txt
+@@ -19,39 +19,6 @@ add_vpp_library(vatplugin
+   LINK_LIBRARIES vppinfra
+ )
+ 
+-##############################################################################
+-# vpp_api_test
+-##############################################################################
+-add_vpp_executable(vpp_api_test ENABLE_EXPORTS
+-  SOURCES
+-  api_format.c
+-  main.c
+-  plugin.c
+-  json_format.c
+-  types.c
+-  ip_types_api.c
+-  ip_types.c
+-  protocols.def
+-
+-  DEPENDS api_headers
+-
+-  LINK_LIBRARIES
+-  vlibmemoryclient
+-  svm
+-  vatplugin
+-  vppinfra
+-  Threads::Threads
+-  dl
+-)
+-
+-##############################################################################
+-#  vpp_json_test
+-##############################################################################
+-add_vpp_executable(vpp_json_test ENABLE_EXPORTS NO_INSTALL
+-  SOURCES json_format.c json_test.c
+-  LINK_LIBRARIES vppinfra m
+-)
+-
+ ##############################################################################
+ # vat headers
+ ##############################################################################
+diff --git src/vat2/CMakeLists.txt src/vat2/CMakeLists.txt
+index 108e184b5..a90107617 100644
+--- src/vat2/CMakeLists.txt
++++ src/vat2/CMakeLists.txt
+@@ -14,21 +14,6 @@
+ ##############################################################################
+ # vat2
+ ##############################################################################
+-add_vpp_executable(vat2 ENABLE_EXPORTS
```

Comment:
i stripped out quite a bit here, possibly its not needed and was only building due to incorrect cmake flags, not sure

either way it was throwing errors about duplicate symbols